satellite/repair/repairer/peer.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package repairer
+
+import (
+	"context"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/internal/errs2"
+	"storj.io/storj/internal/version"
+	version_checker "storj.io/storj/internal/version/checker"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/peertls/extensions"
+	"storj.io/storj/pkg/peertls/tlsopts"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/pkg/server"
+	"storj.io/storj/pkg/signing"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/contact"
+	"storj.io/storj/satellite/metainfo"
+	"storj.io/storj/satellite/orders"
+	"storj.io/storj/satellite/overlay"
+	"storj.io/storj/satellite/repair/checker"
+	"storj.io/storj/satellite/repair/queue"
+)
+
+// PeerConfig contains the necessary information to configure a repairer peer.
+type PeerConfig struct {
+	Identity identity.Config
+	Server   server.Config
+
+	Contact contact.Config
+	Overlay overlay.Config
+
+	Metainfo metainfo.Config
+	Orders   orders.Config
+
+	Checker  checker.Config
+	Repairer Config
+
+	Version version_checker.Config
+}
+
+// Peer is the repairer process.
+//
+// architecture: Peer
+type Peer struct {
```

Comment:
i think we want to call this RepairProcess or something similar

Reply:
I renamed it `Repairer` for the moment. @egonelbre thoughts here on the names?